autoload/db.vim

```diff
@@ -519,3 +599,128 @@ function! db#clobber_dbext(...) abort
   endfor
   return opts
 endfunction
+
+function! s:vim_job.callback(job, data) dict abort
+  let self.output .= a:data
+endfunction
+
+function! s:vim_job.exit_cb(job, data) dict abort
+  let self.exit = 1
+  let self.exit_status = a:data
+  return self.call_on_finish_if_closed()
+endfunction
+
+function! s:vim_job.close_cb(channel) dict abort
+  let self.close = 1
+  return self.call_on_finish_if_closed()
+endfunction
+
+function! s:vim_job.call_on_finish_if_closed() abort
+  if self.close && self.exit
+    return self.on_finish(split(self.output, "\n", 1), self.exit_status)
+  endif
+endfunction
+
+function! s:nvim_job_callback(jobid, data, event) dict abort
+  if a:event ==? 'exit'
+    return self.on_finish(self.output, a:data)
+  endif
+  let self.output[-1] .= a:data[0]
+  call extend(self.output, a:data[1:])
+endfunction
+
+function! s:job_run(cmd, on_finish, stdin_file, ...) abort
+  let stdin = get(a:, 1, '')
+  if has('nvim')
+    let jobid = jobstart(a:cmd, {
+          \ 'on_stdout': function('s:nvim_job_callback'),
+          \ 'on_stderr': function('s:nvim_job_callback'),
+          \ 'on_exit': function('s:nvim_job_callback'),
+          \ 'output': [''],
+          \ 'on_finish': a:on_finish,
+          \ })
+
+    if !empty(a:stdin_file) && filereadable(a:stdin_file)
+      let stdin = readfile(a:stdin_file, 'b')
+    endif
+
+    if !empty(stdin)
+      call chansend(jobid, stdin)
+      call chanclose(jobid, 'stdin')
+    endif
+
+    return jobid
+  endif
+
+  if exists('*job_start')
+    let fn = copy(s:vim_job)
+    let fn.on_finish = a:on_finish
+    let opts = {
+          \ 'callback': fn.callback,
+          \ 'exit_cb': fn.exit_cb,
+          \ 'close_cb': fn.close_cb,
+          \ 'mode': 'raw'
+          \ }
+
+    if has('patch-8.1.350')
+      let opts['noblock'] = 1
+    endif
+
+    if !empty(a:stdin_file) && filereadable(a:stdin_file)
+      let opts['in_io'] = 'file'
+      let opts['in_name'] = a:stdin_file
+      let fn.close = 1
+      let stdin = ''
+    endif
+
+    let job = job_start(a:cmd, opts)
+
+    if !empty(stdin)
+      call ch_sendraw(job, stdin)
+      call ch_close_in(job)
+      let fn.close = 1
+    endif
+
+    return job
+  endif
+
+  throw 'DB: jobs not supported by this vim version.'
+endfunction
+
+function! s:job_wait(job, ...) abort
+  let timeout = get(a:, 1, -1)
+
+  if has('nvim')
+    return jobwait([a:job], timeout)[0] == -1 ? v:false : v:true
+  endif
+
+  let finished = v:true
+  if exists('*job_status')
+    let ms = 0
+    let max = timeout
+    while job_status(a:job) ==# 'run'
+      if ms == max
+        let finished = v:false
+        break
+      endif
+      let ms += 1
+      sleep 1m
+    endwhile
```

Comment:
Let's kill the `timeout` argument entirely. It's not used anywhere and the Vim implementation is kind of sus.